src/main/java/seedu/address/logic/parser/patient/DeletePatientCommandParser.java

```diff
@@ -13,18 +16,41 @@
  */
 public class DeletePatientCommandParser implements Parser<DeletePatientCommand> {
 
+    /**
+     * Used for the separation of force delete format and patient index.
+     */
+    private static final Pattern FORCE_DELETE_FORMAT = Pattern.compile("(?<forceDelete>\\D+)(?<patientIndex>.*)");
+
     /**
      * Parses the given {@code String} of arguments in the context of the DeletePatientCommand
      * and returns a DeletePatientCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
     public DeletePatientCommand parse(String args) throws ParseException {
+        final Matcher forceDeleteMatcher = FORCE_DELETE_FORMAT.matcher(args.trim());
+        boolean isForceDelete = false;
+        String indexToParse = args;
+
         try {
```

Comment:
actually, i think a cleaner way may be to check the index first, if index is wrong then throw a parse exception for index related stuf (ie `DeletePatientCommand.MESSAGE_USAGE`, only after index is confirmed to be good, then parse for forceDelete.

this is because currently the emphasis of the parse exception message usage being returned isn't very clear, 
ie if user keys command `delete-patient --force notAnIndex`, this method returns a `ParseException` with `FORCE_DELETE_MESSAGE_USAGE`. Emphasis of the exception should have been on the Index being wrong instead,

Reply:
The reason why I do it this way is so that if the index is wrong for `delete-patient --force notAnIndex`, the parseException will be raised as well. I will add in an invalid index message before the message usage to improve on the clarity of the exception.